### PR TITLE
Add descriptor QA logging

### PR DESCRIPTION
**Pull request by HansKristian-Work**
Combined with UMR, we have a way to debug GPU hangs caused by invalid descriptors.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -232,3 +232,82 @@ void DEBUG_CHANNEL_MSG(float v0, float v1, ...); // Up to 4 components, ...
...
+All descriptor updates and copies are logged so that it's possible to correlate descriptors with
+GPU crash dumps. `enable_descriptor_qa` is not enabled by default,
+since it adds some flat overhead in an extremely hot code path.
+
+### Debugging descriptor crashes with RADV/ACO dumps (hardcore ultra nightmare mode)
+
+For when you're absolutely desperate, there is a way to debug GPU hangs.
+First, install [umr](https://gitlab.freedesktop.org/tomstdenis/umr) and make the binary setsuid.
+
+`ACO_DEBUG=force-waitcnt RADV_DEBUG=hang VKD3D_DESCRIPTOR_QA_LOG=Z:/somewhere/desc.txt %command%`
```
Comment (doitsujin) [inline README.md:248]: Wine does actually understand Unix file paths so just `/somewhere/desc.txt` would work fine as well.

File: libs/vkd3d/resource.c [hunk 2]
```diff
@@ -3511,6 +3535,10 @@ void d3d12_desc_copy(struct d3d12_desc *dst, struct d3d12_desc *src,
         struct vkd3d_bound_buffer_range *dst_buffer_ranges = dst->heap->buffer_ranges.host_ptr;
         dst_buffer_ranges[dst->heap_offset] = src_buffer_ranges[src->heap_offset];
     }
+
+#ifdef VKD3D_ENABLE_DESCRIPTOR_QA
+    vkd3d_descriptor_debug_copy_descriptor(dst->heap, dst->heap_offset, src->heap, src->heap_offset, src->metadata.cookie);
+#endif
```
Comment (doitsujin) [inline libs/vkd3d/resource.c:3541]: This part may need some rebasing since `d3d12_desc_copy` is a different function now.

**Review by doitsujin — CHANGES_REQUESTED** [review]
Needs rebase + one possible readme nit, otherwise LGTM.

File: README.md (hunk 1 above)
Comment (HansKristian-Work) [reply README.md]: Fixed.

File: libs/vkd3d/resource.c (hunk 2 above)
Comment (HansKristian-Work) [reply libs/vkd3d/resource.c:3620]: Fixed.

**Review by doitsujin — APPROVED** [review]

**Pull request merged**